snuba/datasets/discover.py

```diff
@@ -13,37 +13,32 @@
     UUID,
 )
 from snuba.datasets.dataset import TimeSeriesDataset
-from snuba.datasets.dataset_schemas import DatasetSchemas
-from snuba.datasets.errors_replacer import ReplacerState
+from snuba.datasets.events import EventsDataset
 from snuba.datasets.factory import get_dataset
-from snuba.datasets.schemas import Schema, RelationalSource
-from snuba.datasets.transactions import BEGINNING_OF_TIME
+from snuba.datasets.storage import QueryStorageSelector, Storage, TableStorage
+from snuba.datasets.transactions import TransactionsDataset
 from snuba.query.extensions import QueryExtension
 from snuba.query.parsing import ParsingContext
 from snuba.query.project_extension import ProjectExtension, ProjectWithGroupsProcessor
 from snuba.query.query import Query
 from snuba.query.query_processor import QueryProcessor
 from snuba.query.processors.apdex_processor import ApdexProcessor
-from snuba.query.processors.impact_processor import ImpactProcessor
 from snuba.query.processors.basic_functions import BasicFunctionsProcessor
-from snuba.query.processors.prewhere import PrewhereProcessor
-from snuba.query.processors.readonly_events import ReadOnlyTableSelector
-from snuba.query.processors.tagsmap import NestedFieldConditionOptimizer
+from snuba.query.processors.impact_processor import ImpactProcessor
 from snuba.query.timeseries import TimeSeriesExtension
-from snuba.query.types import Condition
 from snuba.request.request_settings import RequestSettings
 from snuba.util import is_condition
 
 EVENTS = "events"
 TRANSACTIONS = "transactions"
 
 
-def detect_dataset(
+def detect_table(
     query: Query, events_columns: ColumnSet, transactions_columns: ColumnSet
 ) -> str:
     """
     Given a query, we attempt to guess whether it is better to fetch data from the
-    "events" or "transactions" dataset. This is going to be wrong in some cases.
+    "events" or "transactions" table. This is going to be wrong in some cases.
```

Comment:
Are "table" and "storage" substitutable here? Is one more accurate?

Reply:
in this specific context there is a 1:1 mapping so we could use the interchangeably. More in general, in today's snuba table=storage though I would hesitate setting this assumptions, it could break simply if we used a clickhouse view.